{
	"id": "RLSA-2021:1761",
	"advisories": [
		{
			"content": {
				"id": "RLSA-2021:1761",
				"title": "Moderate: python27:2.7 security and bug fix update",
				"description": "Python is an interpreted, interactive, object-oriented programming language that supports modules, classes, exceptions, high-level dynamic data types, and dynamic typing. The python27 packages provide a stable release of Python 2.7 with a number of additional utilities and database connectors for MySQL and PostgreSQL.\n\nSecurity Fix(es):\n\n* python: CRLF injection via HTTP request method in httplib/http.client (CVE-2020-26116)\n\n* python-urllib3: CRLF injection via HTTP request method (CVE-2020-26137)\n\n* python-lxml: mXSS due to the use of improper parser (CVE-2020-27783)\n\n* python: Stack-based buffer overflow in PyCArg_repr in _ctypes/callproc.c (CVE-2021-3177)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\n\nAdditional Changes:\n\nFor detailed information on changes in this release, see the Rocky Linux 8.4 Release Notes linked from the References section.",
				"severity": [
					{
						"type": "vendor",
						"source": "errata.rockylinux.org",
						"vendor": "MODERATE"
					}
				],
				"references": [
					{
						"source": "MITRE",
						"url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-26116"
					},
					{
						"source": "MITRE",
						"url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-26137"
					},
					{
						"source": "MITRE",
						"url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-27783"
					},
					{
						"source": "MITRE",
						"url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-3177"
					},
					{
						"source": "Red Hat",
						"url": "https://bugzilla.redhat.com/show_bug.cgi?id=1883014"
					},
					{
						"source": "Red Hat",
						"url": "https://bugzilla.redhat.com/show_bug.cgi?id=1883632"
					},
					{
						"source": "Red Hat",
						"url": "https://bugzilla.redhat.com/show_bug.cgi?id=1886754"
					},
					{
						"source": "Red Hat",
						"url": "https://bugzilla.redhat.com/show_bug.cgi?id=1901633"
					},
					{
						"source": "Red Hat",
						"url": "https://bugzilla.redhat.com/show_bug.cgi?id=1918168"
					},
					{
						"source": "errata.rockylinux.org",
						"url": "https://errata.rockylinux.org/RLSA-2021:1761"
					}
				],
				"published": "2021-05-18T06:02:07Z"
			},
			"segments": [
				{
					"ecosystem": "rocky:8"
				}
			]
		}
	],
	"vulnerabilities": [
		{
			"content": {
				"id": "CVE-2020-26116",
				"severity": [
					{
						"type": "cvss_v31",
						"source": "MITRE",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:N",
							"base_score": 6.5,
							"base_severity": "MEDIUM",
							"temporal_score": 6.5,
							"temporal_severity": "MEDIUM",
							"environmental_score": 6.5,
							"environmental_severity": "MEDIUM"
						}
					}
				],
				"cwe": [
					{
						"source": "MITRE",
						"cwe": [
							"CWE-113"
						]
					}
				],
				"references": [
					{
						"source": "MITRE",
						"url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-26116"
					}
				]
			},
			"segments": [
				{
					"ecosystem": "rocky:8"
				}
			]
		},
		{
			"content": {
				"id": "CVE-2020-26137",
				"severity": [
					{
						"type": "cvss_v31",
						"source": "MITRE",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:N",
							"base_score": 6.5,
							"base_severity": "MEDIUM",
							"temporal_score": 6.5,
							"temporal_severity": "MEDIUM",
							"environmental_score": 6.5,
							"environmental_severity": "MEDIUM"
						}
					}
				],
				"cwe": [
					{
						"source": "MITRE",
						"cwe": [
							"CWE-113"
						]
					}
				],
				"references": [
					{
						"source": "MITRE",
						"url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-26137"
					}
				]
			},
			"segments": [
				{
					"ecosystem": "rocky:8"
				}
			]
		},
		{
			"content": {
				"id": "CVE-2020-27783",
				"severity": [
					{
						"type": "cvss_v31",
						"source": "MITRE",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N",
							"base_score": 6.1,
							"base_severity": "MEDIUM",
							"temporal_score": 6.1,
							"temporal_severity": "MEDIUM",
							"environmental_score": 6.1,
							"environmental_severity": "MEDIUM"
						}
					}
				],
				"cwe": [
					{
						"source": "MITRE",
						"cwe": [
							"CWE-79"
						]
					}
				],
				"references": [
					{
						"source": "MITRE",
						"url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-27783"
					}
				]
			},
			"segments": [
				{
					"ecosystem": "rocky:8"
				}
			]
		},
		{
			"content": {
				"id": "CVE-2021-3177",
				"severity": [
					{
						"type": "cvss_v31",
						"source": "MITRE",
						"cvss_v31": {
							"vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H",
							"base_score": 5.9,
							"base_severity": "MEDIUM",
							"temporal_score": 5.9,
							"temporal_severity": "MEDIUM",
							"environmental_score": 5.9,
							"environmental_severity": "MEDIUM"
						}
					}
				],
				"cwe": [
					{
						"source": "MITRE",
						"cwe": [
							"CWE-120"
						]
					}
				],
				"references": [
					{
						"source": "MITRE",
						"url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-3177"
					}
				]
			},
			"segments": [
				{
					"ecosystem": "rocky:8"
				}
			]
		}
	],
	"detections": [
		{
			"ecosystem": "rocky:8",
			"conditions": [
				{
					"criteria": {
						"operator": "OR",
						"criterions": [
							{
								"type": "version",
								"version": {
									"vulnerable": true,
									"fix_status": {
										"class": "fixed"
									},
									"package": {
										"name": "python-dns",
										"architectures": [
											"src"
										]
									},
									"affected": {
										"type": "rpm",
										"range": [
											{
												"lt": "0:1.15.0-10.el8"
											}
										],
										"fixed": [
											"0:1.15.0-10.el8"
										]
									}
								}
							},
							{
								"type": "version",
								"version": {
									"vulnerable": true,
									"fix_status": {
										"class": "fixed"
									},
									"package": {
										"name": "python-lxml",
										"architectures": [
											"src"
										]
									},
									"affected": {
										"type": "rpm",
										"range": [
											{
												"lt": "0:4.2.3-4.el8"
											}
										],
										"fixed": [
											"0:4.2.3-4.el8"
										]
									}
								}
							},
							{
								"type": "version",
								"version": {
									"vulnerable": true,
									"fix_status": {
										"class": "fixed"
									},
									"package": {
										"name": "python-markupsafe",
										"architectures": [
											"src"
										]
									},
									"affected": {
										"type": "rpm",
										"range": [
											{
												"lt": "0:0.23-19.el8"
											}
										],
										"fixed": [
											"0:0.23-19.el8"
										]
									}
								}
							},
							{
								"type": "version",
								"version": {
									"vulnerable": true,
									"fix_status": {
										"class": "fixed"
									},
									"package": {
										"name": "python-psycopg2",
										"architectures": [
											"src"
										]
									},
									"affected": {
										"type": "rpm",
										"range": [
											{
												"lt": "0:2.7.5-7.el8"
											}
										],
										"fixed": [
											"0:2.7.5-7.el8"
										]
									}
								}
							}
						]
					}
				}
			]
		}
	],
	"data_source": {
		"id": "rocky-errata",
		"raws": [
			"vuls-data-raw-rocky-errata/2021/RLSA-2021:1761.json"
		]
	}
}
